File: extra5/app/src/main/java/com/example/extra5/api/NewsApi.java
package com.example.extra5.api;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface NewsApi {
    @GET("v2/top-headlines")
    Call<NewsResponse> top(@Query("country") String country, @Query("apiKey") String key);

    class NewsResponse {
        public List<ArticleDto> articles;
    }

    class ArticleDto {
        public String title;
        public String description;
        public String urlToImage;
    }
}

File: extra5/app/src/main/java/com/example/extra5/api/NewsApi.kt
package com.example.extra5.api

import retrofit2.http.GET
import retrofit2.http.Query

data class NewsResponse(val articles: List<ArticleDto>)
data class ArticleDto(val title: String?, val description: String?, val urlToImage: String?)

interface NewsApi {
    @GET("v2/top-headlines")
    suspend fun top(@Query("country") country: String, @Query("apiKey") key: String): NewsResponse
}
